src/lib/pure/ketrew_edsl.mli

```diff
@@ -251,3 +252,56 @@ val pbs :
   [> `Long_running of string * string ]
 (** Create a “PSB” build process. *)
 
+
+val yarn_application :
+  ?host:Ketrew_host.t ->
+  ?daemonize_using:[ `Nohup_setsid | `Python_daemon ] ->
+  ?daemon_start_timeout:float ->
+  Ketrew_program.t -> [> `Long_running of string * string ]
+(** Create a build process that requests resources from Yarn, the
+    command must be an application in the Yarn sense (i.e.
+    a program that is going to contact Yarn by itself to request
+    containers):
```

Comment:
I added an issue about it https://github.com/hammerlab/ketrew/issues/120 so we can move on. But in general, I think what I find confusing is that now there are two things which describe a sequence of commands:
1. `call_script`
2. `Build_process.t`

I guess, I understand that we need some kind of starting point (it's not turtles all the way down). But it might be a documentation/refactoring point to specify that.